tests/test_calibration.py

```diff
@@ -376,3 +378,76 @@ def test_conditions_markdown():
     cond = LPDConditions()
     md = MarkdownFormatter()(cond)  # Smoketest
     assert isinstance(md, str)
+
+
+@pytest.mark.vcr
+def test_DetectorData_from_identifier():
+    # SPB-AGIPD, multi-module detector with full CalCat entries
+    agipd = DetectorData.from_identifier('SPB_DET_AGIPD1M-1')
+    repr(agipd)
+
+    assert agipd
+    assert agipd.identifier == 'SPB_DET_AGIPD1M-1'
+    assert agipd.source_names[0] == 'SPB_DET_AGIPD1M-1/DET/0CH0:xtdf'
+    assert agipd.detector_type == 'AGIPD-Type'
+    assert len(agipd) == agipd.number_of_modules
+    assert list(agipd)[0] == next(iter(agipd.keys())) == 'AGIPD00'
+    assert isinstance(next(iter(agipd.values())), DetectorModule)
+    assert agipd[0] == agipd['AGIPD00']
+
+    # PDU
+    pdu = agipd[0]
+    assert pdu.aggregator == 'AGIPD00'
+    assert pdu.ccv_params == (
+        'AGIPD_SIV1_AGIPDV11_M517', 101003000000, 'AGIPD-Type')
+
+    # FXE-JFHZ, single-module detector with partial CalCat entries
+    jfhz = DetectorData.from_identifier('FXE_XAD_JFHZ')
+    repr(jfhz)
+
+    assert jfhz
+    assert len(jfhz) == 1
+    assert jfhz.number_of_modules is None
+
+    with pytest.raises(AssertionError):
+        jfhz.first_module_index
+
+    # SQS-DSSC, PDU-less detector
+    dssc = DetectorData.from_identifier('SQS_DET_DSSC1M-1')
+    repr(dssc)
+
+    assert not dssc
+    assert len(dssc) == 0
+
+    with pytest.raises(ValueError):
+        dssc.detector_type
+
+
+@pytest.mark.vcr
+def test_DetectorData_from_instrument():
+    with pytest.raises(ValueError):
+        DetectorData.from_instrument('SPB')  # More than one detector.
+
+    with pytest.raises(ValueError):
+        DetectorData.from_instrument('SPB', '*JF*')  # More than one JF.
+
+    # Sufficiently narrow glob.
+    jf4m = DetectorData.from_instrument('SPB', '*JF4M')
+    assert jf4m.identifier == 'SPB_IRDA_JF4M'
+
+    # Instrument with single detector.
+    hirex = DetectorData.from_instrument('SA1')
+    assert hirex.identifier == 'SA1_XTD9_HIREX'
+
+
+@pytest.mark.vcr
+def test_DetectorData_list_by_instrument():
+    assert DetectorData.list_by_instrument('SCS') == [
```

Comment:
I don't think this can be used with a snapshot time at the moment, but maybe we can ask if that's easy to add.

Reply:
Yes, I double-checked the detector APIs and indeed they do not have snapshotting in those APIs at the moment.